File: Loenn/effects/heatWaveOneMode.lua
local heatWaveOneMode = {}

heatWaveOneMode.name = "StyleMaskHelper/HeatWaveOneMode"
heatWaveOneMode.canBackground = true
heatWaveOneMode.canForeground = true

heatWaveOneMode.defaultData = {
    coreMode = "Hot",
    colorGrade = false
}

heatWaveOneMode.fieldInformation = {
    coreMode = {
        options = {"Hot", "Cold", "None"},
        editable = false
    }
}

return heatWaveOneMode

File: Ahorn/effects/heatWaveOneMode.jl
module StyleMaskHelperHeatWaveOneMode

using ..Ahorn, Maple

@mapdef Effect "StyleMaskHelper/HeatWaveOneMode" HeatWaveOneMode(only::String="*", exclude::String="", coreMode::String="Hot", colorGrade::Bool=false)

const placements = HeatWaveOneMode

Ahorn.canFgBg(effect::HeatWaveOneMode) = true, true

Ahorn.editingOptions(effect::HeatWaveOneMode) = Dict{String, Any}(
    "coreMode" => ["Hot", "Cold", "None"]
)

end